src/modules/hunting/cvehunter.py

```diff
@@ -0,0 +1,121 @@
+import json
+import requests
+import uuid
+import ast
+
+from ...core.events import handler
+from ...core.events.types import Vulnerability, Event
+from ..discovery.apiserver import ApiServer
+from ...core.types import Hunter, ActiveHunter, KubernetesCluster, RemoteCodeExec, AccessRisk, InformationDisclosure, \
+    PrivilegeEscalation, DenialOfService
+
+""" Vulnerabilities """
+
+
+class ServerApiVersionEndPointAccessPE(Vulnerability, Event):
+    """Node is vulnerable to critical CVE-2018-1002105"""
+
+    def __init__(self, evidence):
+        Vulnerability.__init__(self, KubernetesCluster, name="Critical Privilege Escalation CVE", category=PrivilegeEscalation)
+        self.evidence = evidence
+
+
+class ServerApiVersionEndPointAccessDos(Vulnerability, Event):
+    """Node is vulnerable to critical CVE-2019-1002100"""
+
+    def __init__(self, evidence):
+        Vulnerability.__init__(self, KubernetesCluster, name="Medium Denial of Service CVE", category=DenialOfService)
+        self.evidence = evidence
+
+
+# Passive Hunter
+@handler.subscribe(ApiServer)
+class IsVulnerableToCVEAttack(Hunter):
+    """ Node is running a Kubernetes version vulnerable to critical CVE-2018-1002105 """
+
+    def __init__(self, event):
+        self.event = event
+        self.headers = dict()
+        self.path = "https://{}:{}".format(self.event.host, self.event.port)
+        self.service_account_token_evidence = ''
+        self.api_server_evidence = ''
+        self.k8sVersion = ''
+
+    def get_service_account_token(self):
+        logging.debug(self.event.host)
+        logging.debug('Passive Hunter is attempting to access pod\'s service account token')
+        try:
+            with open('/var/run/secrets/kubernetes.io/serviceaccount/token', 'r') as token:
+                data = token.read()
+                self.service_account_token_evidence = data
+                self.headers = {'Authorization': 'Bearer ' + self.service_account_token_evidence}
+                return True
+        except IOError:  # Couldn't read file
+            return False
+
+    def get_api_server_version_end_point(self):
+        logging.debug(self.event.host)
+        logging.debug('Passive Hunter is attempting to access the API server version end point using the pod\'s service account token')
+        try:
+            res = requests.get("{path}/version".format(path=self.path),
+                               headers=self.headers, verify=False)
+            self.api_server_evidence = res.content
+            resDict = ast.literal_eval(res.content)
+            version = resDict["gitVersion"].split('.')
+            first_two_minor_digits = eval(version[1])
+            last_two_minor_digits = eval(version[2])
+            return [first_two_minor_digits, last_two_minor_digits]
+
+        except (requests.exceptions.ConnectionError, KeyError):
+            return None
+
+    def check_cve_2018_1002105(self, api_version):
+        first_two_minor_digists = api_version[0]
```

Comment:
In an ideal world we'd correct the spelling in these to all be "digits" rather than "digists", but that's just superficial